File: crossecore-java/src/java_/SingleVariableAccessBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class SingleVariableAccessBase 
extends ExpressionImpl implements SingleVariableAccess
{
	private VariableDeclaration _variable;
	public VariableDeclaration getVariable()
	{
		
			return _variable;
		
		
	}
	public void setVariable(VariableDeclaration value){
		if (value != _variable) {
			NotificationChain msgs = null;
			if (_variable != null){
				msgs = ((InternalEObject)_variable).eInverseRemove(this, Java_PackageImpl.VARIABLEDECLARATION_USAGEINVARIABLEACCESS, SingleVariableAccess.class, msgs);
			}
			if (value != null){
				msgs = ((InternalEObject)value).eInverseAdd(this, Java_PackageImpl.VARIABLEDECLARATION_USAGEINVARIABLEACCESS, SingleVariableAccess.class, msgs);
			}
			msgs = basicSetVariable(value, msgs);
			if (msgs != null) {
				msgs.dispatch();
			}
		}
		else if (eNotificationRequired()){
			eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.SINGLEVARIABLEACCESS_VARIABLE , value, value));
		}
	}
	private Expression _qualifier;
	public Expression getQualifier()
	{
		
			return _qualifier;
		
		
	}
	public void setQualifier(Expression value){
		if (value != _qualifier) {
			NotificationChain msgs = null;
			if (_qualifier != null){
				msgs = ((InternalEObject)_qualifier).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.SINGLEVARIABLEACCESS_QUALIFIER, null, msgs);
			}
			if (value != null){
				msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.SINGLEVARIABLEACCESS_QUALIFIER, null, msgs);
			}
			msgs = basicSetQualifier(value, msgs);
			if (msgs != null) {
				msgs.dispatch();
			}
		}
		else if (eNotificationRequired()){
			eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.SINGLEVARIABLEACCESS_QUALIFIER , value, value));
		}
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.SINGLEVARIABLEACCESS;
	}
	
	@Override
	public NotificationChain eInverseAdd(InternalEObject otherEnd, int featureID, NotificationChain msgs) {
		switch (featureID) {
			case Java_PackageImpl.SINGLEVARIABLEACCESS_VARIABLE:
				if (_variable != null){
					msgs = ((InternalEObject)_variable).eInverseRemove(this, Java_PackageImpl.SINGLEVARIABLEACCESS_VARIABLE, VariableDeclaration.class, msgs);
				}
				return basicSetVariable((VariableDeclaration)otherEnd, msgs);
		}
		return super.eInverseAdd(otherEnd, featureID, msgs);
	}
	
	@Override
	public NotificationChain eInverseRemove(InternalEObject otherEnd, int featureID, NotificationChain msgs) {
		switch (featureID) {
			case Java_PackageImpl.SINGLEVARIABLEACCESS_VARIABLE:
				return basicSetVariable(null, msgs);
		}
		return super.eInverseRemove(otherEnd, featureID, msgs);
	}
	
	public NotificationChain basicSetQualifier(Expression newobj, NotificationChain msgs) {
		Expression oldobj = _qualifier;
		_qualifier = newobj;
		if (eNotificationRequired()) {
			ENotificationImpl notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.SINGLEVARIABLEACCESS_QUALIFIER, oldobj, newobj);
			if (msgs == null){
				msgs = notification;
			}
			else{
				msgs.add(notification);
			}
		}
		return msgs;
	}
	public NotificationChain basicSetVariable(VariableDeclaration newobj, NotificationChain msgs) {
		VariableDeclaration oldobj = _variable;
		_variable = newobj;
		if (eNotificationRequired()) {
			ENotificationImpl notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.SINGLEVARIABLEACCESS_VARIABLE, oldobj, newobj);
			if (msgs == null){
				msgs = notification;
			}
			else{
				msgs.add(notification);
			}
		}
		return msgs;
	}
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.SINGLEVARIABLEACCESS_VARIABLE:
				return getVariable();
			case Java_PackageImpl.SINGLEVARIABLEACCESS_QUALIFIER:
				return getQualifier();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.SINGLEVARIABLEACCESS_VARIABLE:
				setVariable((VariableDeclaration) newValue);
				return;
			case Java_PackageImpl.SINGLEVARIABLEACCESS_QUALIFIER:
				setQualifier((Expression) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.SINGLEVARIABLEACCESS_VARIABLE:
				return getVariable() != null; //single, volatile
			case Java_PackageImpl.SINGLEVARIABLEACCESS_QUALIFIER:
				return _qualifier != null; //single != null;
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-typescript/java_/SingleVariableAccessBase.ts
import {EClass} from "ecore/EClass";
import {Set} from "ecore/Set";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {ExpressionImpl} from "./ExpressionImpl";
import {Expression} from "./Expression";
import {VariableDeclaration} from "./VariableDeclaration";
import {SingleVariableAccess} from "./SingleVariableAccess";
		
			export class SingleVariableAccessBase
			extends ExpressionImpl
			implements SingleVariableAccess
			{
				private _variable:VariableDeclaration = null;
				get variable():VariableDeclaration{
				
					return this._variable;
				}
				set variable(value:VariableDeclaration) {
					if (value != this._variable) {
						let msgs:NotificationChain = null;
						if (this._variable != null){
							msgs = (this._variable).eInverseRemove(this, Java_PackageLiterals.VARIABLEDECLARATION_USAGEINVARIABLEACCESS, /*SingleVariableAccess*/ null , msgs);
						}
						if (value != null){
							msgs = value.eInverseAdd(this, Java_PackageLiterals.VARIABLEDECLARATION_USAGEINVARIABLEACCESS, /*SingleVariableAccess*/ null, msgs);
						}
						msgs = this.basicSetVariable(value, msgs);
						if (msgs != null) {
							msgs.dispatch();
						}
					}
					else if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.SINGLEVARIABLEACCESS_VARIABLE , value, value));
					}
				}
				private _qualifier:Expression = null;
				get qualifier():Expression{
				
					return this._qualifier;
				}
				set qualifier(value:Expression) {
					if (value != this._qualifier) {
						let msgs:NotificationChain = null;
						if (this._qualifier != null){
							msgs = (this._qualifier).eInverseRemove(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.SINGLEVARIABLEACCESS_QUALIFIER, /*null*/ null , msgs);
						}
						if (value != null){
							msgs = value.eInverseAdd(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.SINGLEVARIABLEACCESS_QUALIFIER, /*null*/ null, msgs);
						}
						msgs = this.basicSetQualifier(value, msgs);
						if (msgs != null) {
							msgs.dispatch();
						}
					}
					else if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.SINGLEVARIABLEACCESS_QUALIFIER , value, value));
					}
				}

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return SingleVariableAccessBase.eStaticClass;
				}
			
				public eInverseAdd(otherEnd:InternalEObject, featureID:number, msgs:NotificationChain): NotificationChain{
					switch (featureID) {
						case Java_PackageLiterals.SINGLEVARIABLEACCESS_VARIABLE:
							if (this._variable != null){
								msgs = this._variable.eInverseRemove(this, Java_PackageLiterals.SINGLEVARIABLEACCESS_VARIABLE, /*VariableDeclaration*/ null, msgs);
							}
							return this.basicSetVariable(otherEnd as VariableDeclaration, msgs);
					}
					//return this.eInverseAddFromExpression(otherEnd, featureID, msgs);
					return super.eInverseAdd(otherEnd, featureID, msgs);
				}
				//public eInverseAddFromSingleVariableAccess = this.eInverseAdd;
				
				public eInverseRemove(otherEnd:InternalEObject, featureID:number, msgs:NotificationChain):NotificationChain{
					switch (featureID) {
						case Java_PackageLiterals.SINGLEVARIABLEACCESS_VARIABLE:
							return this.basicSetVariable(null, msgs);
					}
					//return this.eInverseRemoveFromExpression(otherEnd, featureID, msgs);
					return super.eInverseRemove(otherEnd, featureID, msgs);
				}
				
				//public eInverseRemoveFromSingleVariableAccess = this.eInverseRemove;
			
				public basicSetVariable(newobj:VariableDeclaration, msgs:NotificationChain):NotificationChain {
					let oldobj = this._variable;
					this._variable = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.SINGLEVARIABLEACCESS_VARIABLE, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				public basicSetQualifier(newobj:Expression, msgs:NotificationChain):NotificationChain {
					let oldobj = this._qualifier;
					this._qualifier = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.SINGLEVARIABLEACCESS_QUALIFIER, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.SINGLEVARIABLEACCESS_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.SINGLEVARIABLEACCESS_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.SINGLEVARIABLEACCESS_ORIGINALCLASSFILE:
							return this.originalClassFile;
						case Java_PackageLiterals.SINGLEVARIABLEACCESS_VARIABLE:
							return this.variable;
						case Java_PackageLiterals.SINGLEVARIABLEACCESS_QUALIFIER:
							return this.qualifier;
					}
					//return this.eGetFromExpression(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.SINGLEVARIABLEACCESS_VARIABLE:
							this.variable = <VariableDeclaration> newValue;
							return;
						case Java_PackageLiterals.SINGLEVARIABLEACCESS_QUALIFIER:
							this.qualifier = <Expression> newValue;
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
